# IGV 2.10.3 - Jul 2021

* Fixed a problem with displaying the popup information in alignment tracks - no details were shown for paired-end alignments when ***View as pairs*** was enabled. [Git issue #1003](http://github.com/igvteam/igv/issues/1003) 
 
# IGV 2.10.2 - Jul 2021

**Bug fixes:**

* Fixed a problem with displaying extra large chromosomes (1024^3 bp). [Git issue #977](http://github.com/igvteam/igv/issues/977) (initial issue) 

* Fixed a problem with loading BAM files that do not store read sequence. [Git issue #977](http://github.com/igvteam/igv/issues/977) (part 2) 

* Tracks generated via Tools > Find Motif are now correctly saved in sessions. [Git issue #988](http://github.com/igvteam/igv/issues/988) 

* Fixed a problem in alignment tracks that caused the right-click popup menu command Copy read details to the clipboard to fail.

* Fixed a problem that sometimes occurred when reloading tracks (via ***File > Reload Tracks***) if using a local reference genome.

# IGV 2.10.0 - Jun 2021

**New features:**

* Support for displaying RNA and DNA base modification tags in BAM files, described in [samtools/hts-specs Git issue #362](https://github.com/samtools/hts-specs/issues/362). These tags are now implemented by several Oxford Nanopore tools that detect base modifications, including Megalodon and Guppy. [Git issue #945](https://github.com/igvteam/igv/issues/945)

* Allow filtering of feature/annotation tracks based on attributes, via ***Tracks > Filter Tracks***. [Git issue #941](http://github.com/igvteam/igv/issues/941)

* New batch command: ```sortByAttribute```

**Bug fixes:**

* Couldn't open *.igv* files from pre-signed Amazon S3 links. [Git issue #922](http://github.com/igvteam/igv/issues/922)

* Tracks were not autoscaling when loading from a port or batch script. [Git issue #925](http://github.com/igvteam/igv/issues/925)

* Changing to autoscale or group autoscale on numeric tracks would not take effect until the user did something to cause a redraw, for example, pan or zoom the view.

* Fixed issue with gene track translations. When the translation table was changed to "vertebrate mitochondrial" and then back to "standard", the translations in the RefSeq track were incorrect. [Git issue #923](http://github.com/igvteam/igv/issues/923)

