
# IGV 2.13.2 - Jul 2022

This release fixes a problem with Sashimi plots - they were not displaying any splice junctions, only the annotation track.

# IGV 2.13.1 - Jun 2022

This release includes the following improvements and fixes:

*   Major performance improvement for CRAM files (htsjdk 3.0).

*   CSI indexes can now be loaded from URLs (htsjdk 3.0).  (Git Issue [#843](https://github.com/igvteam/igv/issues/843))

*   New menu item for feature tracks to specify the feature property to use for the displayed name.  (Git Issue [#1089](https://github.com/igvteam/igv/issues/1089))

*   Improved support for BigBed files; extra fields are now included in the popup text.

*   _Group-by-strand_ now works for feature tracks in collapsed mode.  (Git Issue [#1151](https://github.com/igvteam/igv/issues/1151))

*   Prevent jumping beyond the end of the chromosome.  (Git Issue [#1150](https://github.com/igvteam/igv/issues/1150))



# IGV 2.13.0 - May 2022

This release includes the following improvements and fixes:

*   Support for Amazon AWS default credentials. This enables loading data from AWS S3 using s3:// URLs if Amazon credentials can be found in the default credential chain as described in AWS documentation [here](https://docs.aws.amazon.com/sdk-for-java/v1/developer-guide/credentials.html). In addition, if default credentials are found, the default region chain will be searched as described in AWS documentation [here](https://docs.aws.amazon.com/sdk-for-java/v1/developer-guide/java-dg-region-selection.html). Cognito configuration, as described at [https://umccr.org/blog/igv-amazon-frontend-setup/](https://umccr.org/blog/igv-amazon-frontend-setup/), will take precedence over default credentials.

*   Support for [bigGenePred file format](https://genome.ucsc.edu/goldenPath/help/bigGenePred.html) (Git Issue [#1130](https://github.com/igvteam/igv/issues/1130))

*   Support for [bigNarrowPeak file format](https://genome.ucsc.edu/goldenPath/help/bigNarrowPeak.html).

*   Improved support for jumping to the next/previous feature in a selected feature track (using the "f" and "b" keys).
    *   The feature is now centered in the view.
    *   A preference can now be set to automatically zoom the view to fit the feature to the window, with optional up- and downstream padding. Go to the _General_ tab in _View > Preferences_ to set this preference.
    *   A bug has been fixed where typing an "f" or "b" into text fields (e.g. the locus search box) was interpreted as a feature jump if a feature track was selected. (Git Issue [#1122](https://github.com/igvteam/igv/issues/1122))

*   Support for handling a trailing "." or "?" character in the SAM tag MM (modified bases). (Git Issue [#1131](https://github.com/igvteam/igv/issues/1131))

*   Support for GWAS files with "NA" in the p-value column, e.g. some files produced by new versions of PLINK. (Git Issue [#1127](https://github.com/igvteam/igv/issues/1127))

*   When loading session files that included the _showGenotypes_ property for a VCF track, the track would be displayed without the genotypes even when the property was set to "true". This has now been fixed.

