# IGV 2.17.4 - Mar 2024

**Bug fixes**

* Fixed a performance issue with BigWig files.  GitHub issue [#1505](https://github.com/igvteam/igv/issues/1505).

* Fixed a rendering issue that occurred when a BAM or TDF file was loaded and the locus was set to a chromosome that had no data. An error message was displayed and the IGV window would sometimes display duplicate elements. 

# IGV 2.17.3 - Mar 2024

**New feature**

* **New menu to load a reference genome directly from UCSC GenArk**. 

	Select `Genomes > Load Genome from UCSC GenArkâ€¦` to bring up a table of all available assemblies. You will also be prompted to select from the available annotation tracks. Once you've made your selections, the genome will be added to the genome dropdown menu and the anntation tracks will be displayed whenever you select the genome from the menu. You can change the selection of annotation tracks via `Genomes > Select GenArk Tracks`.

**Bug fixes**

* When saving a Sashimi plot to an SVG file, it produced an extremely large file. It now produces a much smaller file.

* The chromosome-scale navigation bar was not displayed when there was no cytoband specification, for example, when you load only a FASTA file for the reference genome.  GitHub issue [#1498](https://github.com/igvteam/igv/issues/1498).

* Fixed an issue with validating base modification MM tags on negative strand alignments. GitHub issue [#1476](https://github.com/igvteam/igv/issues/1476).


# IGV 2.17.2 - Feb 2024

**Bug fixes**

* BAM/CRAM files with malformed SA tags were not viewable.

* Validation of base modification tag (MM) was sometimes incorrect for reverse strand alignments. GitHub issue [#1476](https://github.com/igvteam/igv/issues/1476).

* The alignment track "copy clipped sequence" menu item incorrectly copied reverse complement sequence for reverse strand alignments. GitHub issue [#1487](https://github.com/igvteam/igv/issues/1487).


# IGV 2.17.1 - Jan 2024

**New feature**

* **Track hubs**

    UCSC track hubs in the [useOneFile](https://genome.ucsc.edu/goldenPath/help/hgTracksHelp.html#UseOneFile) format can now
    be loaded directly into IGV to define a reference genome and associated tracks. Notably this includes the more than
    3,600 genome assemblies and associated tracks hosted at the [UCSC GenArk](https://hgdownload.soe.ucsc.edu/hubs/) site.
    See the [Reference genome](../UserGuide/reference_genome.md#load-a-track-hub) section for more details on how to use a track hub for IGV's reference genome.
    
    For more information on the GenArk resource see `Clawson, H., Lee, B.T., Raney, B.J. et al. GenArk: towards a million UCSC genome browsers. Genome Biol 24, 217 (2023).`
    [https://doi.org/10.1186/s13059-023-03057-x](https://doi.org/10.1186/s13059-023-03057-x)

**Other updates**

* Popup information text is now selectable.

* Bug fix - RNA splicing gaps between alignment blocks were sometimes drawn incorrectly.  See Github issue [#1470](https://github.com/igvteam/igv/issues/1470)


# IGV 2.17.0 - Jan 2024

!!! Danger "As of version 2.17.0, IGV requires **Java 17 or greater**." 
    If you download one of the IGV versions that does not include Java, make sure you have Java 17 installed and in your path."


**New features**

* Base modifications:

    * Base modification options now include the choice of monocolor and two-color schemes for all modifiction types. Previously the two-color scheme was restricted to 5mC modifications. See [Base modifications](../UserGuide/tracks/alignments/base_modifications.md) in the User Guide for details.

    * Colors for each modification, as well as colors to represent the unmodified base in two-color schemes, can now be customized from a new `Base Mods` tab on the Preferences dialog.

    * Base modifications are now shown for expanded insertions.

    * Properly account for base modifications to "N" in coverage track. Fixes GitHub issue [#1444](https://github.com/igvteam/igv/issues/1444)
    
    * Base modification MM tags are now validated against the count of the modified base in the read sequence. If the MM tag requires more bases than available it is marked invalid and not shown. This validation can be disabled in the user preferences. See GitHub issue [#1435](https://github.com/igvteam/igv/issues/1435) for discussion on this issue.

* Alignment tracks:

    * Expanded insertions have been redesigned. Insertions for all loaded alignment tracks are now indicated by
      triangles under the genomic ruler.
      
    * New group option: by selected insertion locus.
   
    * Chimeric reads - Visualization of chimeric (split) reads have been improved with enhanced display of clipping
      information, as well as a new split read alignment diagram. For details,
      see [Chimeric reads](../UserGuide/tracks/alignments/chimeric_reads.md) in the User Guide.
      
    * New graphical annotations for INDEL quality on SAM/BAM/CRAM files containing Ultima Genomics flow-based reads.
      For details, see [Ultima indel coloring](../UserGuide/tracks/alignments/ultima/ultima.md) in the User Guide. For more information
      on Ultima flow-based sequencing
      see their [preprint in bioRxiv](https://www.biorxiv.org/content/10.1101/2022.05.29.493900)


* Autosaved sessions - IGV sessions can now be automatically saved and restored.
  See [Sessions](../UserGuide/sessions.md) in the User Guide for more information.


* New file formats:

    * GVCF (
      see [description](https://gatk.broadinstitute.org/hc/en-us/articles/360035531812-GVCF-Genomic-Variant-Call-Format)
      on the GATK site).
      
    * bedMethyl (
      see [description](https://www.google.com/url?sa=t&rct=j&q=&esrc=s&source=web&cd=&cad=rja&uact=8&ved=2ahUKEwjnurf9zfmCAxU9FFkFHfjeAwsQFnoECA4QAw&url=https%3A%2F%2Fwww.encodeproject.org%2Fdata-standards%2Fwgbs%2F%23%3A~%3Atext%3Dstates%2520at%2520CpG.-%2CDescription%2520of%2520bedMethyl%2520file%2CStart%2520position%2520in%2520chromosome&usg=AOvVaw21Dwl3k4lFCnoVxG8q8Ffg&opi=89978449)
      on the ENCODE site).

**Other updates**

* Added missing blank line to HTTP 204 response. See GitHub pull request [#1368](https://github.com/igvteam/igv/pull/1368).
    
* Increase BLAT sequence limit from 8kb to 25kb to support long read sequences.
    
* New port command `currentGenomePath` to get full path for the currently loaded
      genome. See GitHub issue [#1360](https://github.com/igvteam/igv/pull/1360).
      
* Moved the item *Export track names* from the right-click popup menu to the main *Tracks* menu.
    
* Allow reading VCF4.3 format files with VCF4.4 headers.
    
* New option in the *General* Preferences tab to scale fonts by an arbitrary factor. See
      GitHub issue [#1373](https://github.com/igvteam/igv/issues/1373).



