# IGV 2.17.0 - Jan 2023

**Note:**  IGV now requires Java 17 or greater. If you download one of the IGV versions that does not include Java, make
sure you have Java 17 installed and in your path.

**Whats Changed**:

* Base modifications

    * Base modification options now include the choice of monocolor and two-color schemes for all modifiction types.
      Previously the two-color scheme was restricted to 5mC modifications.
      See [Base modifications](/UserGuide/tracks/alignments/base_modifications/) for full details.

    * Colors for each modification, as well as colors to represent the unmodified base in two-color schemes, can now be
      customized from a new `Base Mods` tab on the Preferences dialog.

    * Base modifications are now shown for expanded insertions.

    * Properly account for base modifications to "N" in coverage track. Fixes #1444
    * Base modification MM tags are now validated against the count of the modified base in the read sequence. If the MM
      tag requires more bases than available it is marked invalid and not shown. This validation can be disabled in the
      user preferences. See [#1435](https://github.com/igvteam/igv/issues/1435) for discussion on this issue.


* Alignment tracks:

    * Expanded insertions have been redesigned. Insertions for all loaded alignment tracks are now indicated by
      triangles under the genomic ruler.
    * New group option: by selected insertion locus
    * Chimeric reads - Visualization of chimeric (split) reads have been improved with enhanced display of clipping
      information, as well as a new split read alignment diagram. For details
      see [Chimeric Reads](/UserGuide/tracks/alignments/chimeric_reads).
    * New graphical annotations for INDEL quality on SAM/BAM/CRAM files containing Ultima Genomics flow-based reads.
      For details see [Ultima indel coloring](/UserGuide/tracks/alignments/ultima/ultima/). For more information
      on Ultima flow-based sequencing
      see [https://www.biorxiv.org/content/10.1101/2022.05.29.493900](https://www.biorxiv.org/content/10.1101/2022.05.29.493900)

* Autosaved sessions - IGV sessions can now be automatically saved and restored.
  See [Sessions](/UserGuide/sessions/#session-autosave) for more information.


* New file formats:
    * GVCF (
      see [description](https://gatk.broadinstitute.org/hc/en-us/articles/360035531812-GVCF-Genomic-Variant-Call-Format)
      on the GATK site).
    * bedMethyl (
      see [description](https://www.google.com/url?sa=t&rct=j&q=&esrc=s&source=web&cd=&cad=rja&uact=8&ved=2ahUKEwjnurf9zfmCAxU9FFkFHfjeAwsQFnoECA4QAw&url=https%3A%2F%2Fwww.encodeproject.org%2Fdata-standards%2Fwgbs%2F%23%3A~%3Atext%3Dstates%2520at%2520CpG.-%2CDescription%2520of%2520bedMethyl%2520file%2CStart%2520position%2520in%2520chromosome&usg=AOvVaw21Dwl3k4lFCnoVxG8q8Ffg&opi=89978449)
      on the ENCODE site).


* Miscellaneous

    * Added missing blank line to http 204 response [#1368](https://github.com/igvteam/igv/pull/1368).
    * Increase BLAT sequence limit from 8kb to 25kb to support long read sequences.
    * New port command `currentGenomePath` to get full path for the currently loaded
      genome [#1360](https://github.com/igvteam/igv/pull/1360).
    * Moved "export track names" from popup to "Tracks" menu.
    * Allow reading VCF4.3 format files with VCF4.4 headers.
    * New option in **General** Preferences tab to scale fonts by an arbitrary factor. See
      issue [#1373](https://github.com/igvteam/igv/issues/1373).




