# IGV 2.18.0 - July 2024

**New features**

* If a file does not have a recognizable filename extenstion, IGV now tries to determine the file format from the file contents. This works for BAM, CRAM, BigWig, BigBed, VCF, GFF3, and TDF content. [GitHub issue #1526](https://github.com/igvteam/igv/issues/1526).
* New display mode for alignment tracks: `Full` displays a single alignment per row. [GitHub issue #1531](https://github.com/igvteam/igv/issues/1531).
* New right-click menu option to remove empty data panels. 

**Bug Fixes**

* Wait cursor could hang in some situations. 
* Chromosome order was not maintained when using FASTA files to define the reference genome. [GitHub issue #1489](https://github.com/igvteam/igv/issues/1489).
* Now can automatically detect
* Restored missing BAM files to Encode 2012 menus.

**Other updates**

* Third Gen alignment preferences:
    * Quick consensus mode is now FALSE by default. [GitHub issue #1516](https://github.com/igvteam/igv/issues/1516).
    * Hiding small indels is now FALSE by default.

<p></p>

* Long-read RNA-seq alignments are now assigned an experiment type of `RNA` by default. The experiment type can be changed to `3rd Gen` via the track's right-click popup menu. [GitHub issue #1491](https://github.com/igvteam/igv/issues/1491).

* Ultima Genomics alignment data. [GitHub issue #1490](https://github.com/igvteam/igv/pull/1490).
    * Fixed several edge cases and inconsistencies in the coloring of data from Ultima Genomics.
    * Added a new option to the Alignment preferences: `INDEL coloring uses min quality (ultima)` colors indels by the minimal quality of the homopolymer run rather than by the quality of the specific change.
    * Thanks to Thomas Garrety (Hartwig Foundation) for insightful comments and feature suggestions.

    